### PR TITLE
Add s390x support for travis.

### PR DESCRIPTION
**Pull request by srajmane**
As Travis CI [officially supports](https://blog.travis-ci.com/2019-11-12-multi-cpu-architecture-ibm-power-ibm-z) s390x builds, adding support for same.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -38,8 +41,19 @@ matrix:
...
+      apt:
+        packages:
+        - texlive-latex-extra
+        - texlive-fonts-recommended
   - env: CI_KIND=build XARCH=x64 CONFIG_ARG=--disable-shared
   - env: CI_KIND=build XARCH=x64 MIN_BUILD=1
+  - env: CI_KIND=build XARCH=s390x CONFIG_ARG=--disable-shared
+    arch: s390x
+  - env: CI_KIND=build XARCH=s390x MIN_BUILD=1
+    arch: s390x
```
Comment (dra27) [inline .travis.yml:56]: There's no need for either of these tests - the `MIN_BUILD` and `--disable-shared` tests ensure that a section of the build system works correctly, it's not necessary to run them on multiple architectures

**Review by dra27 — COMMENTED** [review]
Thanks for looking at this - however, I'm not convinced this is necessary for OCaml. We already test s390 on Inria's main CI system and pull requests are tested if deemed worthwhile (these are references to "precheck"). I'm not sure that adding additional jobs to our Travis builds (and so reducing the number of simultaneous tests we can run) adds enough benefit.

If others disagree, it's not certainly not necessary to do more than the basic and flambda tests (and probably not worth the extra flambda test, either)

File: tools/ci/travis/travis-ci.sh [hunk 2]
```diff
@@ -118,6 +118,9 @@ EOF
       AS='as' ASPP='gcc -c' \
       $configure_flags
     ;;
+  s390x)
```
Comment (dra27) [inline tools/ci/travis/travis-ci.sh:121]: (this doesn't need to be a separate case - just make the `x64)` above `x64|s390x)`

**Comment by nojb** [comment]
This will make the CI runs take even longer (they already take a really long time). Given that this system is already tested by INRIA's CI, I don't see a good reason to go ahead with this PR.

**Comment by xavierleroy** [comment]
I agree with @dra27 and @nojb.

The s390x port is frequently tested as part of Inria's CI, using a VM kindly provided by IBM Toronto. So far it's been a very reliable port: if it works with x86/Linux, it works with s390x.  So, I don't think we would gain much assurance by testing every pull request on s390x via Travis. And I'm afraid this would be a waste of Travis' s390x VMs, which are probably less abundant than x86 VMs.

For these reasons I'm closing this PR.

**Comment by cwsolee** [comment]
Hi @xavierleroy, @dra27 & @nojb  , First of all, thanks for working with my team for all these years, the VM is provided by my team and I think you probably get a email from your contact(Bill) about migration of that VM to a newer version of hardware. Your contact person will be changed from Bill to Snehal, she is the one that submit this PR too. Now that Travis is supporting z officially, should we move to Travis instead and you don't need to go thru the VM migration either. I'm ok if VM is preferred, I just thought it's a good timing to do the change.

**Comment by xavierleroy** [comment]
It's a small world :-)  For the time being, I'd rather keep the zSystems VM (and migrate it whenever convenient for you): not only it is part of our CI at Inria, but it's also very useful to test changes interactively via ssh.  Thanks for you help!

**Comment by cwsolee** [comment]
No problem, let us know if u would like to add z on travis later and keeping the new VM is certainly ok, we do provide this type of VM for debugging on z purpose also. Snehal will contact you when the new VM is ready, should be very soon.

**Pull request closed without merging**